components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -231,10 +231,20 @@ class AddonsManagerAdapter(
     internal fun fetchIcon(addon: Addon, iconView: ImageView, scope: CoroutineScope = this.scope): Job {
         return scope.launch {
             try {
+                // We calculate how much time takes to fetch an icon,
+                // if takes less than a second, we assume it comes
+                // from a cache and we don't show any transition animation.
+                val startTime = System.currentTimeMillis()
```

Comment:
I could find a better way to identify if icon comes from a cache, if you have any alternative please let me know  :)

Reply:
Heh, interesting solution to prevent the flickering if the icon comes in later / too late!